container/config.py

```diff
@@ -78,9 +76,12 @@ def _resolve_defaults(self, config):
         defaults = config.setdefault('defaults', ordereddict.ordereddict())
         if self.var_file:
             defaults.update(self._get_variables_from_file(), relax=True)
-        defaults.update(self._get_environment_variables(), relax=True)
-        logger.debug(u'Template variables:\n %s' % json.dumps(defaults,
-                                                              indent=4))
+        logger.debug('The default type is', defaults=str(type(defaults)), config=str(type(config)))
+        if type(defaults) == ordereddict.ordereddict:
```

Comment:
Unfortunately it cannot, because `isinstance` also catches instances of `ruamel.yaml.CommentedMapping` which can't use the `relax` kwarg to `.update`. I have to check for the exact class in this case.